packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -0,0 +1,1409 @@
+                      });
+                    },
+                  ),
+                  const Expanded(
+                    child: Text('body'),
+                  ),
+                ],
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    // Tap the second destination.
+    await tester.tap(find.byIcon(Icons.bookmark_border));
+    expect(currentIndex, 1);
+
+    // The second destination animates in.
+    expect(_labelOpacity(tester, 'Def'), equals(0.0));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(_labelOpacity(tester, 'Def'), equals(0.5));
+    await tester.pumpAndSettle();
+    expect(_labelOpacity(tester, 'Def'), equals(1.0));
+
+    // Tap the third destination.
+    await tester.tap(find.byIcon(Icons.star_border));
+    expect(currentIndex, 2);
+
+    // The second destination animates out quickly and the third destination
+    // animates in.
+    expect(_labelOpacity(tester, 'Ghi'), equals(0.0));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 25));
+    expect(_labelOpacity(tester, 'Def'), equals(0.5));
+    expect(_labelOpacity(tester, 'Ghi'), equals(0.0));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 25));
+    expect(_labelOpacity(tester, 'Def'), equals(0.0));
+    expect(_labelOpacity(tester, 'Ghi'), equals(0.0));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(_labelOpacity(tester, 'Ghi'), equals(0.5));
+    await tester.pumpAndSettle();
+    expect(_labelOpacity(tester, 'Ghi'), equals(1.0));
+  });
+
+  testWidgets('Semantics - labelType=[none]', (WidgetTester tester) async {
+    final SemanticsTester semantics = SemanticsTester(tester);
+
+    await _pumpLocalizedTestRail(tester, labelType: NavigationRailLabelType.none);
+
+    expect(semantics, hasSemantics(_expectedSemantics(), ignoreId: true, ignoreTransform: true, ignoreRect: true));
+
+    semantics.dispose();
+  });
+
+  testWidgets('Semantics - labelType=[selected]', (WidgetTester tester) async {
+    final SemanticsTester semantics = SemanticsTester(tester);
+
+    await _pumpLocalizedTestRail(tester, labelType: NavigationRailLabelType.selected);
+
+    expect(semantics, hasSemantics(_expectedSemantics(), ignoreId: true, ignoreTransform: true, ignoreRect: true));
+
+    semantics.dispose();
+  });
+
+  testWidgets('Semantics - labelType=[all]', (WidgetTester tester) async {
+    final SemanticsTester semantics = SemanticsTester(tester);
+
+    await _pumpLocalizedTestRail(tester, labelType: NavigationRailLabelType.all);
+
+    expect(semantics, hasSemantics(_expectedSemantics(), ignoreId: true, ignoreTransform: true, ignoreRect: true));
+
+    semantics.dispose();
+  });
+
+  testWidgets('Semantics - extended', (WidgetTester tester) async {
+    final SemanticsTester semantics = SemanticsTester(tester);
+
+    bool extended = false;
+    StateSetter stateSetter;
```

Comment:
Could the rail not just start in the extended state for this test?

Reply:
I tried this and can't remember why I abandoned the approach. Let me give it another shot

Reply:
Got it working. Nice call!